### PR TITLE
#15: Migrated to Github Actions from Travis

### LINKED ISSUE #4: Add sonar support
## Situation

We use sonar cloud to statically analyze the codebases. Sonar smoothly integrates with Java projects, add to Scala projects.

## Acceptance Criteria

- Check maintained sbt plugin for sonar integration
- Set up the sonar integration

### LINKED ISSUE #15: Migrate to Github actions
## Situation
We should migrate the CI to Github actions.

## Acceptance Criteria

- Moved to GHA
- Updated dependencies

### PR DESCRIPTION
**Pull request by morazow**
Fixes #4
Fixes #15

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: CI Build
+
+on:
+  - push
```
Comment (AnastasiiaSergienko) [inline .github/workflows/ci-build.yml:4]: Do you want to run it also when you merge into main?

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # This required because of the sonarcloud-github-action docker volume mapping.
+      - name: Prepare for Sonar Cloud Scan
+        run: |
+          find . -name scoverage.xml -exec sed -i \
+          's#/home/runner/work/import-export-udf-common-scala/import-export-udf-common-scala#/github/workspace#g' {} +
+
+      - name: Sonar Cloud Scan
+        uses: sonarsource/sonarcloud-github-action@master
```
Comment (AnastasiiaSergienko) [inline .github/workflows/ci-build.yml:58]: Does it run only once or with each version in matrix?

**Review by AnastasiiaSergienko — APPROVED** [review]
Approved with two small comments

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (morazow) [reply .github/workflows/ci-build.yml:4]: Yes, but I think it is run with this setting. Merge to main also creates a new commit push, if I am not wrong. Or did you have something else in mind?

File: .github/workflows/ci-build.yml (hunk 2 above)
Comment (morazow) [reply .github/workflows/ci-build.yml:58]: I think it runs for both, added fix to only run for 2.13.

**Pull request merged**